squidb-processor/src/com/yahoo/squidb/processor/plugins/PluginEnvironment.java

```diff
@@ -212,47 +242,120 @@ private void processPlugin(String pluginName) {
+        unsupportedOptions.removeAll(SQUIDB_SUPPORTED_OPTIONS);
+
+        if (AptUtils.isEmpty(unsupportedOptions)) {
+            return;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (String option : unsupportedOptions) {
+            if (sb.length() > 0) {
+                sb.append(SEPARATOR);
+            }
+            sb.append(option);
+        }
+        String message = String.format(UNSUPPORTED_OPTIONS_WARNING, sb);
+        utils.getMessager().printMessage(Diagnostic.Kind.WARNING, message);
+    }
+
+    /**
+     * Returns whether the value of the "squidbOptions" environment option contains the given string
+     *
      * @param option the option to check
      * @return true if the option is set, false otherwise
+     * @deprecated use {@link #hasSquidbOption} instead
      */
+    @Deprecated
     public boolean hasOption(String option) {
+        return hasSquidbOption(option);
+    }
+
+    /**
+     * Returns whether the value of the "squidbOptions" environment option contains the given string
+     *
+     * @param option the option to check
+     * @return true if the option is set, false otherwise
+     * @deprecated use #removeOnGlobalLayoutListener instead
+     */
+    public boolean hasSquidbOption(String option) {
         return squidbOptions.contains(option);
     }
 
     /**
-     * @return envOptions map of annotation processing options obtained from {@link ProcessingEnvironment#getOptions()}.
+     * Returns the map of annotation processing options obtained from {@link ProcessingEnvironment#getOptions()}.
      * Plugins may access this map to check for their own custom options.
+     * <p>
+     * Note: Custom plugins should also be annotated with {@link SupportedOptions @SupportedOptions} to declare which
+     * environment options keys they support; these will be reported to the toolchain accordingly.
+     *
+     * @return the map of annotation processing options
+     *
+     * @see {@link #hasEnvOption(String)}
      */
     public Map<String, String> getEnvOptions() {
         return envOptions;
     }
 
     /**
-     * @return an AptUtils instance that provides useful annotation processing utility methods
+     * Returns whether the environment options obtained from {@link ProcessingEnvironment#getOptions()} contains the
+     * given key. This is a useful shortcut for plugins which declare their own options key but only need to check
+     * that the key is present, rather than checking the value for that key.
+     * <p>
+     * Note: Custom plugins should also be annotated with {@link SupportedOptions @SupportedOptions} to declare which
+     * environment options keys they support; these will be reported to the toolchain accordingly.
+     *
+     * @return true if the annotation processing options contains the given key, false otherwise
      */
-    public AptUtils getUtils() {
-        return utils;
+    public boolean hasEnvOption(String key) {
```

Comment:
I think it might be worth adding a `getEnvOptionValue(String key)` method just as a convenience.